### ISSUE TITLE
Allow user to specify number of common dynamic components

### ISSUE THREAD

**Issue by dbrakenhoff** [pastas/metran#11]
It would be a nice feature to be able to override the automatic method to determine the number of common dynamic components. 

For example:
```python
import metran as mt

ml = mt.Metran(oseries, nfactors=2)
ml.solve()
```

Currently the FactorAnalysis class also contains a maxfactors argument that can presumably be used to limit the no. of factors. This is not exposed through the Metran model class however. So perhaps we should also expose this argument in the Metran class?

Additionally it would be nice to test the current implementation for estimating the number of factors on a dataset that results in 2 (or more) common components. 

So in short:
- [ ] Allow manual setting for number of factors
- [ ] Expose maxfactors keyword argument in FactorAnalysis (if this makes sense)
- [ ] Test metran with dataset that results in 2+ common dynamic components
